superdesk/services.py

```diff
@@ -209,6 +211,36 @@ def remove_from_search(self, item):
         """
         return self.backend.remove_from_search(self.datasource, item)
 
+    def update_data_from_json(self, items):
+        success = []
+        for item in items:
+            try:
+                cv = self.find_one(req=None, _id=item["_id"])
+                # update _created and _updated key if keys provided in json
+                if item.get("_created"):
+                    item["_created"] = cv["_created"] if cv else utcnow()
+                if item.get("_updated"):
+                    item["_updated"] = utcnow()
+
+                res = self.post([item]) if not cv else self.patch(cv["_id"], item)
+                if res:
+                    success.append(res)
+            except Exception as ex:
+                raise SuperdeskApiError.badRequestError(str(ex), exception=ex)
+
+        if success:
+            return {
+                "_status": "SUCCESS",
+                "_success": {"code": 200, "_message": "Vocabularies uploaded successfully."},
```

Comment:
would be good not to use `Vocabularies` in those messages, either use the current datasource or just make that more generic

Reply:
my bad, I totally missed it, I will address this one.